GNUmakefile

```diff
@@ -187,7 +189,8 @@ postbuild_targets += tests
 main_targets := $(sort $(main_targets))
 test_targets := $(sort $(test_targets))
 
-targets += $(tool_targets) $(prebuild_targets) $(main_targets) omr_static_lib $(test_targets) ddr
+targets += $(tool_targets) $(prebuild_targets) $(main_targets) omr_static_lib $(test_targets) ddr $(compiler_prereqs)
```

Comment:
My understanding is, this will always enable the compiler prereqs, which is not the right thing. Iff the compiler is enabled, you should be assert that it's prereqs are enabled in the configure script.

(you edit configure.ac, and compile it into configure using autoconf).

Reply:
All of the compiler prereq targets get built by default, and `port` and `thread` can be optionally disabled. Adding `compiler_prereqs` to `targets` was a way to ensure that someone trying to build the Test Compiler or JITBuilder would also trigger building of the port and thread libraries even if they were set to be disabled. However, the issue with this is that it is removing the ability to explicitly disable the building of `port` and `thread`.

After discussing with @rwy0717 , we have decided that it is best not to assert enabling the compiler dependencies in configure.ac mainly for the following reasons:
* we did not think anyone would have a reason to explicitly disable the port and thread libraries if they want to build the Test Compiler and JITBuilder. Doing so would result in an error anyway if there are no port and thread libraries to link.
* Currently, configure.ac does not make any assertions to ensure top-level inter-component dependencies are enabled despite there being dependencies. It does not make sense to add a special assertion for the Test Compiler and JITBuilder.

Therefore, it is reasonably safe to remove `$(compiler_prereqs)` from `targets`. It would then allow explicitly disabling the building of `port` and `thread`, and will be alright if the test compiler and JITBuilder are not explicitly enabled.